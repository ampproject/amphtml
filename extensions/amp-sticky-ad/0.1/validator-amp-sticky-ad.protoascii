#
# Copyright 2016 The AMP HTML Authors. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS-IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the license.
#
tags: {  # <script custom-element="amp-sticky-ad">
  html_format: AMP  # Ads are not allowed inside ads.
  tag_name: "SCRIPT"
  spec_name: "amp-sticky-ad extension .js script"
  mandatory_parent: "HEAD"
  attrs: {
    name: "async"
    mandatory: true
    value: ""
  }
  attrs: {
    name: "custom-element"
    mandatory: true
    value: "amp-sticky-ad"
    dispatch_key: true
  }
  attrs: {
    name: "src"
    mandatory: true
    value_regex: "https://cdn\\.ampproject\\.org/v0/amp-sticky-ad-(latest|0\\.1)\\.js"
  }
  attrs: {
    name: "type"
    value_casei: "text/javascript"
  }
  cdata: {
    blacklisted_cdata_regex: {
      regex: "."
      error_message: "contents"
    }
  }
  spec_url: "https://www.ampproject.org/docs/reference/extended/amp-sticky-ad.html"
}
tags: {  # <amp-sticky-ad>
  html_format: AMP
  tag_name: "AMP-STICKY-AD"
  disallowed_ancestor: "AMP-APP-BANNER"
  disallowed_ancestor: "AMP-SIDEBAR"
  also_requires_tag: "amp-sticky-ad extension .js script"
  attr_lists: "extended-amp-global"
  unique: true
  child_tags: {
    mandatory_num_child_tags: 1
    first_child_tag_name_oneof: "AMP-AD"
  }
  spec_url: "https://www.ampproject.org/docs/reference/extended/amp-sticky-ad.html"
  amp_layout: {
    supported_layouts: NODISPLAY
  }
}
