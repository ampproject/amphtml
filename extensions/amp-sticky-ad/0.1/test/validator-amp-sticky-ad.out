PASS
amp-sticky-ad/0.1/test/validator-amp-sticky-ad.html:40:7 The tag 'amp-ad extension .js script' is missing or incorrect, but required by 'amp-ad'. This will soon be an error. (see https://www.ampproject.org/docs/reference/amp-ad.html) [DEPRECATION]
